source/adapters/cuda/physical_mem.cpp

```diff
@@ -0,0 +1,57 @@
+//===--------- physical_mem.cpp - CUDA Adapter ----------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "physical_mem.hpp"
+#include "common.hpp"
+#include "context.hpp"
+#include "event.hpp"
+
+#include <cassert>
+#include <cuda.h>
+
+UR_APIEXPORT ur_result_t UR_APICALL urPhysicalMemCreate(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice, size_t size,
+    [[maybe_unused]] const ur_physical_mem_properties_t *pProperties,
+    ur_physical_mem_handle_t *phPhysicalMem) {
+  CUmemAllocationProp AllocProps = {};
+  AllocProps.location.type = CU_MEM_LOCATION_TYPE_DEVICE;
+  AllocProps.type = CU_MEM_ALLOCATION_TYPE_PINNED;
+  UR_CHECK_ERROR(GetDeviceOrdinal(hDevice, AllocProps.location.id));
+
+  CUmemGenericAllocationHandle ResHandle;
+  UR_CHECK_ERROR(cuMemCreate(&ResHandle, size, &AllocProps, 0));
```

Comment:
Looks like we need to special case this error code mapping for `cuMemCreate` to line up with the UR spec.

```suggestion
  switch (auto Result = cuMemCreate(&ResHandle, size, &AllocProps, 0)) {
  case CUDA_ERROR_INVALID_VALUE:
    return UR_RESULT_ERROR_INVALID_SIZE;
  default:
    UR_CHECK_ERROR(Result);
  }
```